### PR TITLE
ts: Remove explicit project file for ts parser

### PR DESCRIPTION
**Pull request by MrTibbles**
## Changes
- Removes explicit reference to tsconfig for the @typescript-eslint/parser plugin

## Notes
It appears then when importing the lint library and attempting to override the `parserOptions` field to add additional tsconfig files in the project field it ignores these overrides and uses the base config, that being `@img-arena/eslint-config/ts-rules`. This will be a breaking change as it all consumers of the package will need to add what is being removed to their `eslintrc` files.
For context, this allows for multiple tsconfig files to be used when linting source code which allows us to lint additional directories to "src", for example the cypress directory for fixture files.

### CODE REVIEW

File: ts-rules.js [hunk 1]
```diff
@@ -6,9 +6,6 @@ module.exports = {
     "prettier/@typescript-eslint"
   ],
   parser: "@typescript-eslint/parser",
-  parserOptions: {
-    project: "./tsconfig.json"
```
Comment (parkjh) [inline ts-rules.js:10]: lol, this was the original problem???!!!! 🤦

Comment (parkjh) [reply ts-rules.js:10]: I'm all for this assuming that this package is versioned correctly so that it doesn't impact other projects?

**Review by parkjh — APPROVED** [review]

File: ts-rules.js (hunk 1 above)
Comment (MrTibbles) [reply ts-rules.js:10]: Well, it becamse the problem when we were attempting to pass an array to the field at project level, as in when attempting to override certain settings.
```
parserOptions: {
    noWatch: true,
    projects: ['./tsconfig.json', './cypress/tsconfig.json'],
  },
```

Comment (MrTibbles) [reply ts-rules.js:10]: And yes, its standard npm semver, so this will be a major release

**Pull request merged**